src/vivarium_public_health/metrics/disability.py

```diff
@@ -120,16 +121,27 @@ def disability_weight_aggregator(self, dw: pd.DataFrame) -> float:
 
     def report(
         self,
+        cause_state: Optional[DiseaseState],
         measure: str,
         results: pd.DataFrame,
     ) -> None:
-        """Combine the measure-specific observer results and save to a single file."""
-        measure, cause = [s.strip("_") for s in measure.split("due_to")]
+        """Combine each observation's results and save to a single file"""
+        measure = "ylds"
+        if not cause_state:
+            entity_type = "cause"
+            entity = "all_causes"
+            sub_entity = None
+        else:
+            entity_type = cause_state.cause_type
+            entity = cause_state.model
+            sub_entity = cause_state.state_id
         write_dataframe_to_parquet(
-            measure,
-            results,
-            self.results_dir,
-            self.random_seed,
-            self.input_draw,
-            {COLUMNS.CAUSE: cause},
+            results=results,
+            measure=measure,
+            entity_type=entity_type,
+            entity=entity,
+            sub_entity=sub_entity,
+            results_dir=self.results_dir,
+            random_seed=self.random_seed,
+            input_draw=self.input_draw,
```

Comment:
Is it cleaner to build a dictionary `kwargs` and then do `write_dataframe_to_parquet(**kwargs)`?

Reply:
I'm pretty indifferent. Slightly less lines, though I find inline if/else statements harder to read than actual if: else: blocks